### PR TITLE
Replace styled() with css() in grid

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Updates GUUI's Grid components to meet new code style guidelines

## Why?

Consistency and clarity

### CODE REVIEW

File: packages/guui/components/Grid/index.js [hunk 1]
```diff
@@ -66,37 +65,37 @@ const calculateWidth = (breakpoint: Breakpoint, colspan: number): number => {
...
 
-const ColsStyled = styled('div')(
-    ({ tablet, desktop, leftCol, wide }): MQStyles => {
-        const tabletStyles = gridStyles('tablet', tablet);
-        const desktopStyles = gridStyles('desktop', desktop);
-        const leftColStyles = gridStyles('leftCol', leftCol);
-        const wideStyles = gridStyles('wide', wide);
+const row = css`
+    ${css(clearFix)};
+    margin-left: ${-gutter};
```
Comment (GHaberis) [inline packages/guui/components/Grid/index.js]: It's missing the "px" here, eg: `${-gutter}px`

File: packages/guui/components/Grid/index.js [hunk 2]
```diff
@@ -110,49 +109,33 @@ const normaliseProps = (props: BreakpointProps | number): BreakpointProps => {
...
-
-    return <GridRow>{children}</GridRow>;
-};
-
-Row.defaultProps = {
-    htmlTag: 'div',
-};
+export const Row = ({ children }: { children: React.Node }) => (
+    <div className={row}>{children}</div>
+);
```
Comment (GHaberis) [inline packages/guui/components/Grid/index.js]: So we're removing the ability to define the element type of `Row`? I'm not sure we use it now anyway so maybe this is OK and we can revisit if we find we need that functionality again.

Comment (GHaberis) [reply packages/guui/components/Grid/index.js]: I see we've done the same with `Cols`. I remember the use case which lead to the introduction of `htmlTag` was something like what if we had a `<ul>` which you wanted to be the Row and each `<li>` would be a column. With this implementation this wouldn't be possible.

Comment (SiAdcock) [reply packages/guui/components/Grid/index.js]: Yeah, you’re right to flag this up, I was just thinking the same thing. I’ll reinstate the feature using React.createElement

File: packages/guui/components/Grid/index.js [hunk 3]
```diff
@@ -114,13 +113,19 @@ export const Row = ({
...
-    children: React$Node,
-}) => {
-    const GridRow = RowStyled.withComponent(htmlTag);
-
-    return <GridRow>{children}</GridRow>;
-};
+    htmlTag: string,
+    children: React.Node,
+}) => (
+    <>
```
Comment (nicl) [inline packages/guui/components/Grid/index.js]: What is this magic?!

File: packages/guui/components/Grid/index.js [hunk 4]
```diff
@@ -44,7 +41,9 @@ const columns: GridSizes = {
     },
 };
 
-const breakpointMqs = {
+const breakpointMqs: {
+    [Breakpoint]: string,
```
Comment (nicl) [inline packages/guui/components/Grid/index.js]: Out of interest, the Breakpoint type represents one of a group of string values here. This is great for explicitness. Does it add any type safety with Flow over just using `string`? I.e. is there any way to leverage the additional typing we've added here?

File: packages/guui/components/Grid/index.js (hunk 3 above)
Comment (SiAdcock) [reply packages/guui/components/Grid/index.js]: [Fragment](https://reactjs.org/docs/fragments.html)

I'm not 100% sure this is a legit use of fragments, but I couldn't think of another way of defining a component with a dynamic tag.

File: packages/guui/components/Grid/index.js (hunk 4 above)
Comment (SiAdcock) [reply packages/guui/components/Grid/index.js]: It prevents the keys of this object from being anything other than one of the `Breakpoint` strings

```js
// good
const breakpointMqs = {
  tablet: '@media(min-width: 740px)'
}

// bad
const breakpointMqs = {
  foo: '@media(min-width: 740px)'
}
```

I'm not sure if there's any real type safety advantage to it? Maybe  like you say, it's good for explicitness.

Comment (nicl) [reply packages/guui/components/Grid/index.js]: Ah yep - no that makes sense. Thanks.

**Pull request merged**